service/src/main/java/bio/terra/workspace/service/resource/referenced/flight/update/UpdateReferenceResourceFlight.java

```diff
@@ -0,0 +1,39 @@
+package bio.terra.workspace.service.resource.referenced.flight.update;
+
+import bio.terra.stairway.Flight;
+import bio.terra.stairway.FlightMap;
+import bio.terra.workspace.common.utils.FlightBeanBag;
+import bio.terra.workspace.common.utils.RetryRules;
+import bio.terra.workspace.service.job.JobMapKeys;
+import bio.terra.workspace.service.resource.referenced.ReferencedResource;
+import bio.terra.workspace.service.resource.referenced.flight.create.ValidateReferenceStep;
+
+/** A flight to update the reference resource's name, description, and/or attributes. */
+public class UpdateReferenceResourceFlight extends Flight {
+
+  /**
+   * Flight to update a reference resource target.
+   *
+   * @param inputParameters FlightMap of the inputs for the flight
+   * @param beanBag Anonymous context meaningful to the application using Stairway
+   */
+  public UpdateReferenceResourceFlight(FlightMap inputParameters, Object beanBag) {
+    super(inputParameters, beanBag);
+
+    FlightBeanBag appContext = FlightBeanBag.getFromObject(beanBag);
+    final ReferencedResource resource =
+        inputParameters.get(JobMapKeys.REQUEST.getKeyName(), ReferencedResource.class);
+
+    // Perform access verification
+    addStep(new ValidateReferenceStep(appContext), RetryRules.cloud());
+
+    addStep(
+        new RetrieveReferenceMetadataStep(
+            appContext.getResourceDao(), resource.getWorkspaceId(), resource.getResourceId()),
+        RetryRules.shortDatabase());
```

Comment:
nit: the methods like `RetryRules.shortDatabase()` creates a RetryRule object. The same object can be used for more than one step in a flight (but cannot be used between flights). So you could create one shortDatabase retry rule and use it in both this step and the next one. It is a minor efficiency.